webapp/src/hooks/useRelayerBalanceInfo.ts

```diff
@@ -0,0 +1,53 @@
+import { type UseQueryResult, useQuery } from "@tanstack/react-query";
+import { ethers } from "ethers";
+import type { SettingsFormData } from "@/components/settings/SettingsForm";
+import { useSession } from "@/contexts/SessionContext";
+import { useHarbourRpcProvider } from "@/hooks/useRpcProvider";
+
+type UseRelayerBalanceInfoProps = Pick<Partial<SettingsFormData>, "rpcUrl">;
+
+interface RelayerBalanceInfo {
+	formatted: string;
+	needsFunding: boolean;
+	faucet?: string;
+}
+
+/**
+ * Custom React Query hook to fetch the current relayer balance information.
+ */
+function useRelayerBalanceInfo(
+	currentSettings: UseRelayerBalanceInfoProps,
+): UseQueryResult<RelayerBalanceInfo | null, Error> {
+	const { keys } = useSession();
+	const { provider } = useHarbourRpcProvider(currentSettings);
+
+	return useQuery<RelayerBalanceInfo | null, Error>({
+		queryKey: [
+			"relayerBalance",
+			keys?.relayer?.address,
+			currentSettings?.rpcUrl,
+		],
+		queryFn: async () => {
+			if (!provider || !keys) {
+				return null;
+			}
+
+			const { chainId } = await provider.getNetwork();
```

Comment:
Should the logic in this queryFn (or potentially the whole queryfn)  be a function in `lib`?